### PR TITLE
[fix] Fixed create_ping and create_config migrations #205

### LINKED ISSUE #205: [migrations] check 0003_create_ping.py is flawed
While migrating the system used at the University of Urbino by @cappe87 and @AlessiaVe, we incurred in a flaw in this data migration: 

https://github.com/openwisp/openwisp-monitoring/blob/8ff5b28f2d5db587c50f0680c0994fbce173b2d0/openwisp_monitoring/check/migrations/0003_create_ping.py#L6-L14

The problem is that `auto_create_ping` assumes we can use the concrete model, but this is not possible on older versions, because some fields are missing and trying to exeucte that will raise an exception.

A possible solution could be to add the possibility of passing the `Check` model as argument, if supplied, this model should be used and the concrete model should not be loaded:

https://github.com/openwisp/openwisp-monitoring/blob/8ff5b28f2d5db587c50f0680c0994fbce173b2d0/openwisp_monitoring/check/tasks.py#L37-L54

### PR DESCRIPTION
**Pull request by nepython**
Checks:

- [x] I have manually tested the proposed changes
- [ ] I have written new test cases to avoid regressions (if necessary)
- [ ] I have updated the documentation (e.g. README.rst)

Closes #205

### CODE REVIEW

File: openwisp_monitoring/check/tasks.py [hunk 1]
```diff
@@ -57,17 +65,19 @@ def auto_create_ping(model, app_label, object_id):
     if has_check:
         return
     ct = ContentType.objects.get(app_label=app_label, model=model)
-    check = Check(name='Ping', check=ping_path, content_type=ct, object_id=object_id)
+    check = Check(
+        name='Ping', check=ping_path, content_type_id=ct.id, object_id=object_id
```
Comment (nepython) [inline openwisp_monitoring/check/tasks.py:69]: I am not sure why content_type didn't work with `apps.get_model('check', 'Check')` :/
I was getting this strange error, which I couldn't make understand. Anyways, content_type_id works and we achieve the same objective as prior.
```python
  File "/home/hardik/openwisp/openwisp-monitoring/openwisp_monitoring/check/migrations/0003_create_ping.py", line 15, in create_device_ping
    check_model=Check,
  File "/home/hardik/.local/lib/python3.6/site-packages/celery/local.py", line 191, in __call__
    return self._get_current_object()(*a, **kw)
  File "/home/hardik/.local/lib/python3.6/site-packages/celery/app/task.py", line 392, in __call__
    return self.run(*args, **kwargs)
  File "/home/hardik/openwisp/openwisp-monitoring/openwisp_monitoring/check/tasks.py", line 66, in auto_create_ping
    check = Check(name='Ping', check=ping_path, content_type=ct, object_id=object_id)
  File "/home/hardik/.local/lib/python3.6/site-packages/django/db/models/base.py", line 482, in __init__
    _setattr(self, field.name, rel_obj)
  File "/home/hardik/.local/lib/python3.6/site-packages/django/db/models/fields/related_descriptors.py", line 219, in __set__
    self.field.remote_field.model._meta.object_name,
ValueError: Cannot assign "<ContentType: config | device>": "Check.content_type" must be a "ContentType" instance.
```

File: openwisp_monitoring/check/tasks.py [hunk 2]
```diff
@@ -57,17 +65,19 @@ def auto_create_ping(model, app_label, object_id):
     if has_check:
         return
     ct = ContentType.objects.get(app_label=app_label, model=model)
```
Comment (nemesisdesign) [inline openwisp_monitoring/check/tasks.py:67]: can you load ContentType with the migration framework as well please?
And test it again.

Comment (nepython) [reply openwisp_monitoring/check/tasks.py]: It worked, thanks :+1:

**Comment by nepython** [comment]
> How did you test this?

I started with a clean db, created admin and then ran migrations. After all the migrations were run, I simply ran the below command to unmigrate,
```shell
./tests/manage.py migrate check 0002
```
Now I opened django shell, created four new devices.
Next step, ran the below command to test the fix,
```shell
./tests/manage.py migrate check
```
So everything forward and reverse migrations worked out fine for me. I am using `postgresql` with `postgis` as the relational database.

Also I once unmigrated and migrated back to apply your suggestion related to passing `ContentType` model, then I was using this very database so I think it is works fine even without a clean db for me.

**Comment by nemesisdesign** [comment]
> > How did you test this?
> 
> I started with a clean db, created admin and then ran migrations. After all the migrations were run, I simply ran the below command to unmigrate,
> 
> ```shell
> ./tests/manage.py migrate check 0002
> ```
> 
> Now I opened django shell, created four new devices.
> Next step, ran the below command to test the fix,
> 
> ```shell
> ./tests/manage.py migrate check
> ```
> 
> So everything forward and reverse migrations worked out fine for me. I am using `postgresql` with `postgis` as the relational database.
> 
> Also I once unmigrated and migrated back to apply your suggestion related to passing `ContentType` model, then I was using this very database so I think it is works fine even without a clean db for me.

Thanks for the explanation.

**Pull request merged**